src/diffusers/models/unets/unet_2d_blocks.py

```diff
@@ -549,21 +554,22 @@ def __init__(
         dropout: float = 0.0,
         num_layers: int = 1,
         resnet_eps: float = 1e-6,
-        resnet_time_scale_shift: str = "default",  # default, spatial
+        resnet_time_scale_shift: str = "default",  # default, spatial, scale_shift
         resnet_act_fn: str = "swish",
         resnet_groups: int = 32,
         attn_groups: Optional[int] = None,
         resnet_pre_norm: bool = True,
         add_attention: bool = True,
         attention_head_dim: int = 1,
+        attention_legacy_order: bool = False,
         output_scale_factor: float = 1.0,
     ):
         super().__init__()
         resnet_groups = resnet_groups if resnet_groups is not None else min(in_channels // 4, 32)
         self.add_attention = add_attention
 
         if attn_groups is None:
-            attn_groups = resnet_groups if resnet_time_scale_shift == "default" else None
+            attn_groups = None if resnet_time_scale_shift == "spatial" else resnet_groups
```

Comment:
this will break for `resnet_time_scale_shift = "scale_shift"`, no? Even though the doc string says only accept `"default"` and `"spatial"` I'm not sure if no model has configured with `resnet_time_scale_shift = scale_shift`

I want to know your thoughts here @sayakpaul

Reply:
Can we maybe come back with a better condition that doesn't leave any grounds for a potentially backward-breaking change? I agree with @yiyixuxu here.

Reply:
> Can we maybe come back with a better condition that doesn't leave any grounds for a potentially backward-breaking change? I agree with @yiyixuxu here.

Cool, let me see if I can solve it in some other way. Would you prefer the current solution, or a slightly more ugly one like a new class of UNetMiddleBlock? In fact, I can easy make it free of backward-breaking, but I might introduce more ugly additional blocks with more code.